src/main/java/hudson/remoting/ResourceImageInJar.java

```diff
@@ -43,11 +53,24 @@
         return new FutureAdapter<byte[],URL>(_resolveJarURL(channel)) {
             @Override
             protected byte[] adapt(URL jar) throws ExecutionException {
+                final URLConnection c;
                 try {
-                    return Util.readFully(toResourceURL(jar,resourcePath).openStream());
+                    URL url = toResourceURL(jar,resourcePath);
+                    c = url.openConnection();
+                    if (DISABLE_FILE_CACHING_IN_JAR_CONNECTION) {
```

Comment:
Why would we not just always disable this?
WIthin the same JVM we should be be called multiple times to adapt the same JAR should we - if we are then would we not get classloading issues?